dango/types/src/account_factory/msg.rs

```diff
@@ -17,27 +18,40 @@ pub struct User {
     pub accounts: BTreeMap<Addr, Account>,
 }
 
-#[grug::derive(Serde)]
-pub struct InstantiateMsg {
+#[grug::derive(Serde, Borsh)]
+pub struct Config {
+    /// The minimum amount of deposit required to register a new user.
+    pub minimum_deposits: Coins,
     /// Code hash to be associated with each account type.
     pub code_hashes: BTreeMap<AccountType, Hash256>,
-    /// Users with associated key to set up during genesis.
-    /// Each genesis user is to be associated with exactly one key.
-    /// A spot account will be created for each genesis user.
-    pub users: BTreeMap<Username, (Hash256, Key)>,
+}
+
+impl Config {
+    pub fn code_hash_for(&self, ty: AccountType) -> anyhow::Result<Hash256> {
+        self.code_hashes
+            .get(&ty)
+            .copied()
+            .ok_or_else(|| anyhow!("code hash not found for account type: {ty}"))
+    }
 }
```

Comment:
I don't like this. Better if we create an abstraction over `BTreeMap` that ensures all enum variants are present in the map keys.